### ISSUE TITLE
Column: FilterMatchModeOptions example

### ISSUE THREAD

**Issue by m-ivanova** [primefaces/primereact#2720]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://forum.primefaces.org/viewforum.php?f=57
```

**Codesandbox Case (Bug Reports)**
https://codesandbox.io/s/primereact-dialog-dropdown-issue-forked-ycg9td?file=/src/App.tsx

**Current behavior**
Can not compile code when trying to use custom filterMatchModeOptions

**Expected behavior**
Code to be compiled without errors and options to be rendered

**Minimal reproduction of the problem with instructions**
Use primereact version 7.2.1 => recreate code from codeSandbox 
Code cant be compiled 

**Please tell us about your environment:**

Versions:
 "primereact": "7.2.0",
    "react": "17.0.2",

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by melloware** [comment]
Isn't' that similar to this old PR: https://github.com/primefaces/primereact/pull/2489/files

I think he shows how to properly use Typescript with Datatable.

**Comment by m-ivanova** [comment]
Okay, could you please show me how to use it. Since it is not a warning but an error, I can not compile the code and therefore can not use this property.

**Comment by melloware** [comment]
try this...

```ts
filterMatchModeOptions={{ label1: "value1", label2: "value2" }}
```

**Comment by m-ivanova** [comment]
I did... no success 

![изображение](https://user-images.githubusercontent.com/35693852/162438149-d281b6d5-e517-4fa5-b0dd-3aa39c34e4df.png)

**Comment by melloware** [comment]
yeah but you were just stating that the compilation failed it was because your key value pairs were not defined correctly

**Comment by melloware** [comment]
You are right I think a showcase example is needed on how to use this.  I don't think this feature is yet complete.

**Comment by mertsincan** [comment]
Hi,
I couldn't replicate it. Could you please try the latest version?
<img width="1776" alt="Screen Shot 2022-04-10 at 17 28 26" src="https://user-images.githubusercontent.com/11868120/162629625-2f72d8e4-c9e5-48f9-8bc1-9f95257fd444.png">

Also, the ts warning issue has been fixed in 8.0.0-rc.2 and Doc has been updated.

For now, closed. If the problem persists, please reopen this issue.
Best Regards,
